src/workflows/airqo_etl_utils/datautils.py

```
@@ -932,6 +820,143 @@
+            data.dropna(
+                subset=dropna_subset,
+                how="all",
+                inplace=True,
+            )
+            data.dropna(subset=["timestamp"], inplace=True)
+            data["timestamp"] = pd.to_datetime(data["timestamp"])
+        except Exception as e:
+            logger.exception(
+                f"There is an issue with the timestamp column. Shape of data: {data.shape}"
+            )
+            raise KeyError(f"An error has occurred with the 'timestamp' column: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)